Framework/DataHandling/src/SaveGDA.cpp

```diff
@@ -0,0 +1,260 @@
+#include "MantidDataHandling/SaveGDA.h"
+
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidAPI/WorkspaceProperty.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/Unit.h"
+#include "MantidKernel/make_unique.h"
+
+#include <boost/optional.hpp>
+
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <sstream>
+
+namespace Mantid {
+namespace DataHandling {
+
+using namespace API;
+
+namespace { // helper functions
+
+const int POINTS_PER_LINE = 4;
+
+double mean(const std::vector<double> &values) {
+  return std::accumulate(values.begin(), values.end(), 0.0) /
+         static_cast<double>(values.size());
+}
+
+double computeAverageDeltaTByT(const HistogramData::HistogramX &tValues) {
+  std::vector<double> deltaTByT;
+  deltaTByT.reserve(tValues.size() - 1);
+
+  std::adjacent_difference(tValues.begin(), tValues.end(),
+                           std::back_inserter(deltaTByT),
+                           [](const double previous, const double current) {
+                             return (previous - current) / current;
+                           });
+  // First element is just first element of tValues, so remove it
+  deltaTByT.erase(deltaTByT.begin());
+  return mean(deltaTByT);
+}
+
+std::string generateBankHeader(int bank, int minT, size_t numberBins,
+                               double deltaTByT) {
+  std::stringstream stream;
+  const auto numberLines = (int)std::ceil((double)numberBins / POINTS_PER_LINE);
+
+  stream << std::setprecision(2) << "BANK " << bank << " " << numberBins << "  "
+         << numberLines << " RALF  " << minT << "  96  " << minT << " "
+         << deltaTByT << " ALT";
+  return stream.str();
+}
+
+boost::optional<std::vector<std::string>>
+getParamLinesFromGSASFile(const std::string &paramsFilename) {
+  std::ifstream paramsFile;
+  paramsFile.open(paramsFilename);
+
+  if (paramsFile.is_open()) {
+    std::vector<std::string> paramLines;
+    std::string line;
+    while (std::getline(paramsFile, line)) {
+      if (line.find("ICONS") != std::string::npos) {
```

Comment:
Comment explaining why ICONS is significant

Or even better a comment and a variable name e.g.:

```
// Icons is used as a delimiter in foo
const std::string delimiter = "ICONS"

if (line.find(delimiter) ...)
```